apps/microsoft-teams/frontend/src/components/config/NotificationEditMode/NotificationEditMode.tsx

```diff
@@ -60,24 +61,22 @@ const NotificationEditMode = (props: Props) => {
         contentTypeId: notification.contentTypeId,
       };
 
-      const { response } = await sdk.cma.appActionCall.createWithResponse(
+      const { result, error } = await sdk.cma.appActionCall.createWithResult(
         {
           appActionId: 'msteamsSendTestMessage',
           environmentId: sdk.ids.environment,
           spaceId: sdk.ids.space,
           appDefinitionId: sdk.ids.app!,
-          userId: sdk.ids.user,
         },
         {
           parameters,
         }
       );
-      const body = JSON.parse(response.body);
 
-      if (body.ok) {
+      if (result && isSendTestMessageResult(result) && result.ok) {
         sdk.notifier.success(editModeFooter.testSuccess);
-      } else {
-        throw new Error(`${editModeFooter.testError}: ${body.error.message}`);
+      } else if (error) {
```

Comment:
By using an `else if` here we have a non exhaustive conditional: That is, it's possible e.g. for us to have a result but not eg result.ok, and no error, thus leading to a silent failure.

I'd recommend we either:

* include an `else` and throw a generic error if we get there (since it's quite likely an application error at that point)
* handle the other cases within the result explicitly (eg. what should we do if the test message result is nto right).

Reply:
Yeah thinking about this more, we should use the `sys.status` as our check here for more reliable behavior.

Also, we should refactor https://github.com/contentful/contentful-management.js/blob/64a9660487d56f2e4d8fd8b7120acb4fb8b0f5f9/lib/entities/app-action-call.ts#L37 to use a discriminating union type. Technically the type definition we've provided here is "incorrect" since it allows for invalid representations. We can fix this and clean your code way up.